In [1]:
%load_ext autoreload
%autoreload 2
from markov import *

# Making of the add_string() method

In [21]:
model = MarkovModel(5)
model.add_string("CS 477 rocks")
model.get_prefixs()
model.get_prefix_count()

CS 47
S 477
 477 
477 r
77 ro
7 roc
 rock
rocks
ocksC
cksCS
ksCS 
sCS 4


12

### Checking actual data

In [26]:
model = MarkovModel(5)
model.load_file_lines("text/spongebobquotes.txt")
print(model.get_prefix_count())
print(model.get_prefix(' you '))

4032
{'s': 4, 'm': 1, 't': 4, 'l': 1, 'j': 1, 'w': 1, 'a': 4, 'b': 1, 'h': 1, 'c': 4, 'k': 1, 'e': 1, 'g': 1}


`{'s': 4, 'm': 1, 't': 4, 'l': 1, 'j': 1, 'w': 1, 'a': 4, 'b': 1, 'h': 1, 'c': 4, 'k': 1, 'e': 1, 'g': 1}`

Should have 4032 unique prefixs;

`' you '` will have all the characters above in the dict

# Computing Probabilities (10 Pts)